packages/vite-node/src/client.ts

```diff
@@ -325,6 +326,25 @@ export class ViteNodeRunner {
     return await this.cachedRequest(id, fsPath, callstack)
   }
 
+  private async _fetchModule(id: string, importer?: string) {
+    try {
+      return await this.options.fetchModule(id)
+    }
+    catch (cause: any) {
+      // rethrow vitest error if it cannot load the module because it's not resolved
+      if (typeof cause?.message === 'string' && cause.message.includes('Failed to load url')) {
```

Comment:
Can this also check Vite's error code `ERR_LOAD_URL` to be more precise?
https://github.com/vitejs/vite/blob/5b0fb047b90d62b6bbabf0cc6ef03b3faa9a4810/packages/vite/src/node/server/transformRequest.ts#L334

Reply:
Added an additional check